endpoints-framework/src/main/java/com/google/api/server/spi/discovery/DiscoveryGenerator.java

```diff
@@ -214,6 +216,21 @@ private void writeApiMethod(ApiConfig config, String servicePath, RestDescriptio
     methods.put(parts.get(parts.size() - 1), method);
   }
 
+  private List<String> computeParameterOrder(ApiMethodConfig methodConfig) {
+    ImmutableSortedSet.Builder<String> queryParamBuilder = ImmutableSortedSet.naturalOrder();
+    Collection<String> pathParameters = methodConfig.getPathParameters();
+    List<String> order = new ArrayList<>(pathParameters);
```

Comment:
Why not use an ImmutableList here too?

Reply:
Also, since `order` is not used in the loop, consider moving it below the loop.

Reply:
It's not used because we have to clone the whole discovery doc at times, and immutable structures are not cloneable. And done.